### PR TITLE
Loading button

### PR DESCRIPTION
**Pull request by Jexeones24**
### Status
**READY**

### Description
- Adds loading `Cta` 
    - loading time is based on `loadingTime` prop
- Extracts `Link` into separate component

[CH2244](https://app.clubhouse.io/hiclark/story/2244/ui-create-loading-and-success-button-update-on-session-cancel-confirmation-modal)

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -28,7 +27,6 @@ export default {
     postcss({
       modules: true,
     }),
-    url(),
```
Comment (Jexeones24) [inline rollup.config.js]: Don't remember _exactly_ why I had to remove this initially, but it prevented the rendering of svgs for some reason.

Comment (willisplummer) [reply rollup.config.js]: feel like we looked at this together but yeah it's hard to remember lol...

all the other svg's in this library still work okay?

File: src/cta/index.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        variant={variant}
+      >
+        <Left>{secondaryIcon && buttonState === 'ready' && secondaryIcon}</Left>
+        <Label>
+          {buttonState === 'ready' ? (
+            children
+          ) : (
+            <Icon
+              clarkSecondary={
+                variant === 'outline' && buttonState === 'confirmed'
```
Comment (willisplummer) [inline src/cta/index.js]: having a hard time understanding what this conditional is about - can we pull it into a const at the beginning of the render method?

like from what i've skimmed, my understanding is that the icon is colored clarkSecondary on success if the button type is outline but i feel like confused kind of i guess and maybe naming could help

File: src/cta/index.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    const { children, disabled, margin, secondaryIcon, variant } = this.props;
+
+    return (
+      <StyledButton
+        disabled={disabled || buttonState === 'loading'}
+        margin={margin}
+        onClick={this.onClickHandler}
+        variant={variant}
+      >
+        <Left>{secondaryIcon && buttonState === 'ready' && secondaryIcon}</Left>
```
Comment (willisplummer) [inline src/cta/index.js]: i think `{buttonState === 'ready' && secondaryIcon}` should work here + clean things up a little

File: src/cta/index.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+const ICONS = {
+  confirmed: <Check />,
+  loading: <Spinner size={ICON_SIZE} />,
+};
+
+type PropsType = {
+  children: Node,
+  disabled: boolean,
+  loadingTime: number,
```
Comment (willisplummer) [inline src/cta/index.js]: lets make this optional

**Pull request merged**